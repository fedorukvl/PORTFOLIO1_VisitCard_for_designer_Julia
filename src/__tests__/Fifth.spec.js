import React from "react";
import { Fifth } from "../Components/index.js";
import { shallow } from "enzyme";

describe("Tests of <Fifth/> component", () => {
	let fifthComponent;
	beforeEach(() => {
		fifthComponent = shallow(<Fifth />);
	});
	afterEach(() => {
		fifthComponent = null;
	});
	describe("First render of <Fifth/> component", () => {
		it("Renders properly", () => {
			expect(fifthComponent).not.toBeNull();
			expect(fifthComponent.debug()).toMatchSnapshot();
		});
	});

	describe("<Fifth/> component renders elements", () => {
		it("Renders elements properly", () => {
			expect(fifthComponent.debug()).toMatchSnapshot();
		});

		it("Renders elements properly (checking divs)", () => {
			expect(
				fifthComponent.find(".workProgressDiv").children()
			).toHaveLength(3);
		});
	});

	describe("Element`s children of <Fifth/> component render properly", () => {
		it(".leftProgressDiv have children", () => {
			expect(
				fifthComponent.find(".leftProgressDiv").children()
			).toHaveLength(4);
		});

		it(".leftProgressDiv have <h1/> with text `Процесс выполнения корсета` as first children", () => {
			expect(
				fifthComponent.find(".leftProgressDiv").childAt(0).type()
			).toEqual("h1");
			expect(fifthComponent.childAt(0).find("h1").text()).toEqual(
				"Процесс выполнения корсета "
			);
		});
	});
});
