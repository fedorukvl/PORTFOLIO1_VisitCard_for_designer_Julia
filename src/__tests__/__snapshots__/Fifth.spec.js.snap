// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <Fifth/> component <Fifth/> component renders elements Renders elements properly 1`] = `
"<div className=\\"workProgressDiv\\">
  <div className=\\"leftProgressDiv\\">
    <h1>
      Процесс выполнения корсета 
    </h1>
    <img src=\\"13.jpg\\" alt=\\"progress example #1\\" />
    <img src=\\"14.jpg\\" alt=\\"progress example #2\\" />
    <img src=\\"15.jpg\\" alt=\\"progress example #3\\" />
  </div>
  <div className=\\"centerProgressDiv\\">
    <img src=\\"16.jpg\\" alt=\\"result example #1\\" />
    <img src=\\"17.jpg\\" alt=\\"result example #2\\" />
  </div>
  <div className=\\"rightProgressDiv\\">
    <h1>
      Этапы выполнения вытачек по разным методикам
    </h1>
    <img src=\\"18.jpg\\" alt=\\"progress example #4\\" />
    <img src=\\"19.jpg\\" alt=\\"progress example #5\\" />
    <img src=\\"20.jpg\\" alt=\\"progress example #6\\" />
  </div>
</div>"
`;

exports[`Tests of <Fifth/> component First render of <Fifth/> component Renders properly 1`] = `
"<div className=\\"workProgressDiv\\">
  <div className=\\"leftProgressDiv\\">
    <h1>
      Процесс выполнения корсета 
    </h1>
    <img src=\\"13.jpg\\" alt=\\"progress example #1\\" />
    <img src=\\"14.jpg\\" alt=\\"progress example #2\\" />
    <img src=\\"15.jpg\\" alt=\\"progress example #3\\" />
  </div>
  <div className=\\"centerProgressDiv\\">
    <img src=\\"16.jpg\\" alt=\\"result example #1\\" />
    <img src=\\"17.jpg\\" alt=\\"result example #2\\" />
  </div>
  <div className=\\"rightProgressDiv\\">
    <h1>
      Этапы выполнения вытачек по разным методикам
    </h1>
    <img src=\\"18.jpg\\" alt=\\"progress example #4\\" />
    <img src=\\"19.jpg\\" alt=\\"progress example #5\\" />
    <img src=\\"20.jpg\\" alt=\\"progress example #6\\" />
  </div>
</div>"
`;
